source/i2c_io.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * i2c_io.c
+ *
+ *  Created on: Aug. 4, 2021
+ *      Author: Robert Taylor
+ */
+#include <i2c_io.h>
+#include "FreeRTOS.h"
+#include "HL_i2c.h"
+#include "os_semphr.h"
+#include <stdint.h>
+#include "os_task.h"
+
+/** @struct i2Csemphr
+*   @brief Interrupt mode globals
+*
+*/
+static struct i2csemphr
+{
+    SemaphoreHandle_t i2c_mutex;
+    SemaphoreHandle_t i2c_block;
+    bool hadFailure;
+} i2csemphr_t[2U];
+
+void init_i2c_driver() {
+    // these can't fail
+    i2csemphr_t[0].i2c_mutex = xSemaphoreCreateMutex();
+    i2csemphr_t[1].i2c_mutex = xSemaphoreCreateMutex();
+    i2csemphr_t[0].i2c_block = xSemaphoreCreateBinary();
+    i2csemphr_t[1].i2c_block = xSemaphoreCreateBinary();
+    i2csemphr_t[0].hadFailure = false;
+    i2csemphr_t[1].hadFailure = false;
+}
+
+int i2c_Receive(i2cBASE_t *i2c, uint8_t addr, uint16_t size, void *buf) {
+    uint8 ret;
+    uint32 index = i2c == i2cREG1 ? 0U : 1U;
```

Comment:
In some circles the shortcut IF is frowned on... Not a big fan myself, so just saying.

Reply:
In this case is the statement simple enough to be acceptable?

Reply:
In this (simple) case it becomes just a question of preference and style. I just wanted to point out my preference. If the logic was more complex, I'd ask to not use it.